contrib/src/main/scala/akka/stream/contrib/SourceGen.scala

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.contrib
+
+import akka.stream._
+import akka.stream.scaladsl._
+import akka.stream.stage._
+
+/**
+ * Source factory methods are placed here
+ */
+object SourceGen {
+
+  /**
+   * Create a `Source` that will unfold a value of type `S` by
+   * passing it through a flow. The flow should emit a
+   * pair of the next state `S` and output elements of type `E`.
+   * Source completes when the flow completes.
+   */
+  def unfoldFlow[S, E, M](seed: S)(flow: Graph[FlowShape[S, (S, E)], M]): Source[E, M] = {
```

Comment:
benchmark code:
https://github.com/hochgi/akka-stream-contrib/commit/2bd97d6463b938b231a4147fe9d8e7636c7f6b70
with fusing, the overhead is rather small, but when fusing is disabled it is very much noticeable. some sample runs:
```none
seed = 1487492288

wrappedTime: 4.241016340000001 ms
bareTime: 2.6165949900000003 ms

wrappedTime: 3.141815640000001 ms
bareTime: 2.008202889999999 ms

seed = 27

wrappedTime: 2.0119849199999993 ms
bareTime: 1.3814176399999993 ms

wrappedTime: 4.65674252 ms
bareTime: 1.7444834599999999 ms
```